src/shapes/circle.class.ts

```diff
@@ -1,201 +1,200 @@
+        radius = this.radius,
+        startX = fabric.util.cos(start) * radius,
+        startY = fabric.util.sin(start) * radius,
+        endX = fabric.util.cos(end) * radius,
+        endY = fabric.util.sin(end) * radius,
+        largeFlag = angle > 180 ? '1' : '0';
+      svgString = [
+        '<path d="M ' + startX + ' ' + startY,
+        ' A ' + radius + ' ' + radius,
+        ' 0 ', +largeFlag + ' 1', ' ' + endX + ' ' + endY,
+        '" ', 'COMMON_PARTS', ' />\n',
+      ];
+    }
+    return svgString;
+  },
+  /* _TO_SVG_END_ */
 
   /**
    * @private
+   * @param {CanvasRenderingContext2D} ctx context to render on
    */
-  function isValidRadius(attributes) {
-    return (('radius' in attributes) && (attributes.radius >= 0));
-  }
-  /* _FROM_SVG_END_ */
+  _render: function(ctx: CanvasRenderingContext2D): void {
+    ctx.beginPath();
+    ctx.arc(
+      0,
+      0,
+      this.radius,
+      degreesToRadians(this.startAngle),
+      degreesToRadians(this.endAngle),
+      false,
+    );
+    this._renderPaintInOrder(ctx);
+  },
 
   /**
-   * Returns {@link fabric.Circle} instance from an object representation
-   * @static
-   * @memberOf fabric.Circle
-   * @param {Object} object Object to create an instance from
-   * @returns {Promise<fabric.Circle>}
+   * Returns horizontal radius of an object (according to how an object is scaled)
+   * @return {Number}
    */
-  fabric.Circle.fromObject = function(object) {
-    return fabric.Object._fromObject(fabric.Circle, object);
-  };
+  getRadiusX: function(): number {
+    return this.get('radius') * this.get('scaleX');
+  },
+
+  /**
+   * Returns vertical radius of an object (according to how an object is scaled)
+   * @return {Number}
+   */
+  getRadiusY: function(): number {
+    return this.get('radius') * this.get('scaleY');
+  },
+
+  /**
+   * Sets radius of an object (and updates width accordingly)
+   * @param {number} value new radius of the circle
+   * @return {Circle} thisArg
+   */
+  setRadius: function(value: number): typeof Circle{
+    this.radius = value;
+    return this.set('width', value * 2).set('height', value * 2);
+  },
+});
+
+/* _FROM_SVG_START_ */
+/**
+ * List of attribute names to account for when parsing SVG element (used by {@link fabric.Circle.fromElement})
+ * @static
+ * @memberOf fabric.Circle
+ * @see: http://www.w3.org/TR/SVG/shapes.html#CircleElement
+ */
+Circle.ATTRIBUTE_NAMES = fabric.SHARED_ATTRIBUTES.concat('cx cy r'.split(' '));
+
+/**
+ * Returns {@link fabric.Circle} instance from an SVG element
+ * @static
+ * @memberOf fabric.Circle
+ * @param {SVGElement} element Element to parse
+ * @param {Function} [callback] Options callback invoked after parsing is finished
+ * @throws {Error} If value of `r` attribute is missing or invalid
+ */
+Circle.fromElement = function(element: SVGElement, callback: (result: typeof Circle) => void) {
+  const parsedAttributes = fabric.parseAttributes(element, fabric.Circle.ATTRIBUTE_NAMES);
```

Comment:
```suggestion
  const parsedAttributes = fabric.parseAttributes(element, Circle.ATTRIBUTE_NAMES);
```

lookup all fabric.Circle and replace them